waspc/test/Generator/JsImportTest.hs

```diff
@@ -0,0 +1,26 @@
+module Generator.JsImportTest where
+
+import qualified StrongPath as SP
+import Test.Tasty.Hspec
+import Wasp.AppSpec.ExtImport
+import Wasp.Generator.JsImport
+import Wasp.JsImport as JI
+
+spec_GeneratorJsImportTest :: Spec
+spec_GeneratorJsImportTest = do
+  describe "extImportToJsImport" $ do
+    it "makes a JsImport from path to ext code dir, relative dir to ext code dir and ExtImport" $ do
+      extImportToJsImport pathToExtCodeDir relDirToExternalCodeDir extImport
+        `shouldBe` JI.JsImport
+          { JI._path = [SP.relfileP|../ext-src/folder/test.js|],
+            JI._name = JsImportModule "test",
+            JI._importAlias = Nothing
+          }
+  where
+    pathToExtCodeDir = [SP.reldirP|ext-src|]
+    relDirToExternalCodeDir = [SP.reldirP|../|]
+    extImport =
+      ExtImport
+        { name = ExtImportModule "test",
+          path = [SP.relfileP|folder/test.js|]
+        }
```

Comment:
It is slightly weird to put this in `where` because it applies to whole `spec_GeneratorJsImportTest` function, so if you add more tests and describes and so on, it becomes too widely scoped, in the sense of that you have defined test setup in a place that is in scope of other tests also.

So instead, I would move these to be defined via `let` in either the specific test or specific describe, however you estimate they are most likely to be used in the future. In this case probably just best to put it into that specific test, and it can be taken out a level above if needed in the future.

Reply:
Makes sense 👍 I went with `where` instinctively  but `let` makes it easier to scope it properly